circuits/aes-gcm/utils.circom

```diff
@@ -403,15 +382,17 @@ template Contains(n) {
     out <== 1 - someEqual.out;
 }
 
+/// m is the number of arrarys, n is the length of each array
 template ArraySelector(m, n) {
     signal input in[m][n];
-    signal input index;
+    signal input index; 
     signal output out[n];
     assert(index >= 0 && index < m);
 
     signal selector[m];
+    component Equal[m];
     for (var i = 0; i < m; i++) {
-        selector[i] <-- index == i ? 1 : 0;
+        selector[i] <== IsEqual()([index, i]);
         selector[i] * (1 - selector[i]) === 0; 
```

Comment:
i don't think we need this constraint now, IsEqual already has that

Reply:
The one on line 415?

Reply:
yeah